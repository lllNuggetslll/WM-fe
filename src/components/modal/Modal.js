import React, { Fragment } from "react";
import { connect } from "react-redux";
import { bool, object, func } from "prop-types";
import Modal from "react-modal";
import { withRouter } from "react-router";
import styled from "styled-components";
import Loader from "./../loader";
import Map from "./../map";
import { ModalHeader, BusinessHours, BusinessAddress } from "./components";
import { mobileBreakPoint } from "./../../constants/breakPoints";

import { shutterListingData } from "./../../actions/actions";
import { deleteParams } from "./../../utils/urlUtils";

const BusinessDetailsContainer = styled.div`
  margin-top: 15px;
  min-width: 450px;
  max-width: 700px;
  display: flex;
  justify-content: space-between;

  @media only screen and (max-width: ${mobileBreakPoint}) {
    flex-direction: column;
  }
`;

const customStyles = {
  content: {
    top: "50%",
    left: "50%",
    right: "auto",
    bottom: "auto",
    marginRight: "-50%",
    transform: "translate(-50%, -50%)",
    maxWidth: 800,
    minWidth: 450
  }
};

const DisplayModal = ({
  isLoadingListing,
  listing,
  history,
  shutterListingData,
  deleteParams
}) => {
  const onCloseModal = () => {
    const url = deleteParams({
      param: "listing",
      location: history.location.search
    });

    shutterListingData();

    history.push(`?${url}`);
  };

  return (
    <Modal
      isOpen={!!(isLoadingListing || listing)}
      onRequestClose={onCloseModal}
      style={customStyles}
    >
      <Loader isLoaded={!isLoadingListing} />
      {!isLoadingListing && listing && (
        <Fragment>
          <ModalHeader {...{ listing, onCloseModal }} />
          <BusinessDetailsContainer>
            <BusinessAddress {...{ listing }} />
            <BusinessHours hours={listing.business_hours} />
          </BusinessDetailsContainer>
          <Map latitude={listing.latitude} longitude={listing.longitude} />
        </Fragment>
      )}
    </Modal>
  );
};

DisplayModal.defaultProps = {
  deleteParams,
  listing: null
};

DisplayModal.propTypes = {
  isLoadingListing: bool.isRequired,
  listing: object,
  history: object.isRequired,
  shutterListingData: func.isRequired,
  deleteParams: func.isRequired
};

const mapStateToProps = state => {
  return {
    isLoadingListing: state.location.isLoadingListing,
    listing: state.location.listing
  };
};

export { DisplayModal as Modal };

export default connect(mapStateToProps, {
  shutterListingData
})(withRouter(DisplayModal));
