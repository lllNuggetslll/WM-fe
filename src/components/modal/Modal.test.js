import React from "react";
import { shallow } from "enzyme";
import { Modal } from "./Modal";

describe("Modal", () => {
  const defaultProps = {
    isLoadingListing: false,
    listing: {},
    history: {},
    shutterListingData: jest.fn()
  };

  it("should render", () => {
    const wrapper = shallow(<Modal {...defaultProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should not render", () => {
    const closedProps = {
      ...defaultProps,
      isLoadingListing: false,
      listing: null
    };
    const wrapper = shallow(<Modal {...closedProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
