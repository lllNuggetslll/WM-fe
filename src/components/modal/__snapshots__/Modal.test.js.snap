// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should not render 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={0}
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "content": Object {
        "bottom": "auto",
        "left": "50%",
        "marginRight": "-50%",
        "maxWidth": 800,
        "minWidth": 450,
        "right": "auto",
        "top": "50%",
        "transform": "translate(-50%, -50%)",
      },
    }
  }
>
  <Loader
    isLoaded={true}
  />
</Modal>
`;

exports[`Modal should render 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={0}
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "content": Object {
        "bottom": "auto",
        "left": "50%",
        "marginRight": "-50%",
        "maxWidth": 800,
        "minWidth": 450,
        "right": "auto",
        "top": "50%",
        "transform": "translate(-50%, -50%)",
      },
    }
  }
>
  <Loader
    isLoaded={true}
  />
  <ModalHeader
    listing={Object {}}
    onCloseModal={[Function]}
  />
  <styled.div>
    <BusinessAddress
      listing={Object {}}
    />
    <BusinessHours />
  </styled.div>
  <WeedMap />
</Modal>
`;
